GitCommands/RevisionReader.cs

```diff
@@ -218,8 +218,8 @@ private bool TryParseRevision(in ArraySegment<byte> chunk, [NotNullWhen(returnVa
             ReadOnlySpan<byte> array = chunk.AsSpan();
 
             // The first 40 bytes are the revision ID and the tree ID back to back
-            if (!ObjectId.TryParseAsciiHexReadOnlySpan(array[..ObjectId.Sha1CharCount], out ObjectId? objectId) ||
-                !ObjectId.TryParseAsciiHexReadOnlySpan(array.Slice(ObjectId.Sha1CharCount, ObjectId.Sha1CharCount), out ObjectId? treeId))
+            if (!ObjectId.TryParse(array[..ObjectId.Sha1CharCount], out ObjectId? objectId) ||
+                !ObjectId.TryParse(array.Slice(ObjectId.Sha1CharCount, ObjectId.Sha1CharCount), out ObjectId? treeId))
```

Comment:
❔ `.Slice`

Reply:
In a follow up.

Some planned changes:

* Optimize ObjectId compare and ToString(). Could be done in an addition to this PR, but I feel it is better to scope creep separately, even if it is separate. 
The performance improvement is not major, but it is used in many operations on the UI why sub ms improvements are welcome.

RevisionReader improvements (really done prior to this PR, but this PR was easier to create and had bigger effects. One or two PR as it seems:
* Reduce allocation reading revisions, (maybe including changes like this too, depends on the diff). Trying to find if email and names are coded in utf8 or not.
* Use git-log "log size" to simplify log reading. This adds about 6% run time to git log (1.9s to 2.0s) for the linux-100K revisions use case, but decreases copying and allows \0 in the pattern (the latter makes Utf8Parser.TryParse() faster than ParseUnixDateTime() by 35 ms so total is likely faster - hard to measure). May just be a draft if I am not sure. (.NET7 may improve performance too.)